### PR TITLE
Update opportunities, upload opportunity feature picture 

### PR DESCRIPTION
**Pull request by EdenWuyifan**

### CODE REVIEW

File: voluncheer/opportunityboard/models.py
Comment (dbleek) [inline voluncheer/opportunityboard/models.py:1]: As we discussed today, I don't think staffing should be set to null=True, blank=True. This will be addressed when I merge the subcategories branch.

File: voluncheer/opportunityboard/tests_views.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    def test_select_page_loads(self):
+        """Tests select page loads"""
+        response = self.client.get(reverse("select"))
+        self.assertEqual(response.status_code, 200)
+
+    def test_post_an_opportunity_page_loads(self):
+        """Tests post_an_opportunity page loads"""
+        response = self.client.get(reverse("post_an_opportunity"))
+        self.assertIn(response.status_code, [200, 302])
```
Comment (dbleek) [inline voluncheer/opportunityboard/tests_views.py:21]: Why are we allowing for success or redirect? Same for below.

File: voluncheer/opportunityboard/tests_views.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    def test_update_an_opportunity_page_loads(self):
+        """Tests update_an_opportunity page loads"""
+        response = self.client.get(reverse("update_an_opportunity", args=[1]))
+        self.assertIn(response.status_code, [200, 302])
```
Comment (dbleek) [inline voluncheer/opportunityboard/tests_views.py:26]: You might have to create an opportunity in the test setUp for this to work correctly. You could copy it from the models test.

File: voluncheer/opportunityboard/views/postanopportunity.py [hunk 3]
```diff
@@ -2,22 +2,50 @@
...
+    if user.is_anonymous:
+        return redirect("home")
+    if user.is_organization:
+        organization_profile = Organization.objects.get(pk=user)
     if request.method == "POST":
-        form = PostAnOpportunityForm(request.POST, instance=request.user)
+        form = PostAnOpportunityForm(request.POST, request.FILES, instance=request.user)
         if form.is_valid():
             form.save()
             return redirect("home")
```
Comment (dbleek) [inline voluncheer/opportunityboard/views/postanopportunity.py:19]: We should have an `else `branch here in case the form is not valid. I'm not sure if that is just a matter of raising an error or perhaps displaying an error message on the page.

File: voluncheer/opportunityboard/views/postanopportunity.py [hunk 4]
```diff
@@ -2,22 +2,50 @@
...
+    user = request.user
+    if user.is_anonymous:
+        return redirect("home")
+    if user.is_organization:
+        organization_profile = Organization.objects.get(pk=user)
+    if request.method == "POST":
+        form = PostAnOpportunityForm(request.POST, request.FILES, instance=request.user)
+        if form.is_valid():
+            form.update(opportunity_id)
+            return redirect("home")
```
Comment (dbleek) [inline voluncheer/opportunityboard/views/postanopportunity.py:40]: Same as above.

File: voluncheer/voluncheer/settings.py
Comment (dbleek) [inline voluncheer/voluncheer/settings.py:1]: Not sure why these lines were moved down or why we are using `"media"` instead of `MEDIA_URL`. Are these changes necessary?

File: voluncheer/opportunityboard/test_forms.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            "description": "Face your destiny.",
+            "date": timezone.now(),
+            "duration": timedelta(minutes=20),
+            "address_1": "New York, NY",
+            "address_2": "Mean st",
+            "is_published": True,
+        }
+        form = PostAnOpportunityForm(data=data, instance=self.user)
+        self.assertEqual(self.user.opportunity_set.all().count(), 0)
+        self.assertFalse(form.save())
```
Comment (dbleek) [inline voluncheer/opportunityboard/test_forms.py:43]: What does it mean to assert that `form.save()` is `False`? Same with the line in the below that asserts if `form.update(1)` is `False`.

File: voluncheer/opportunityboard/test_forms.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                password="peace_and_justice_for_the_galaxy",
+                type=UserType.ORGANIZATION,
+            ),
+            name="Jedi Council",
+        )
+        self.user.save()
+
+    def test_validation(self):
+        """Test Post An Opportunity validation"""
+        data = {
```
Comment (dbleek) [inline voluncheer/opportunityboard/test_forms.py:31]: This can go into a built-in Django test function called `setUp(self)`, which create an opportunity on the test db for each test to use.

File: voluncheer/opportunityboard/test_forms.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        self.assertFalse(form.save())
+        self.assertEqual(self.user.opportunity_set.all().count(), 1)
+        posted_opportunity = self.user.opportunity_set.all().first()
+        self.assertEqual(posted_opportunity.title, "Jedi Trial")
+        self.assertEqual(posted_opportunity.category, "sports")
+        self.assertEqual(posted_opportunity.description, "Face your destiny.")
+
+    def test_update(self):
+        """Test Update An Opportunity validation"""
+        self.test_validation()
```
Comment (dbleek) [inline voluncheer/opportunityboard/test_forms.py:52]: Using a test in another test seems odd, are you sure we want this?

**Review by dbleek — CHANGES_REQUESTED** [review]
Looks good! I noticed a couple of things I'm not sure about, please take a look and let me know if you have any questions.

File: voluncheer/opportunityboard/test_forms.py (hunk 6 above)
Comment (EdenWuyifan) [reply voluncheer/opportunityboard/test_forms.py:31]: Yeah. I'll try add setUp

File: voluncheer/voluncheer/settings.py
Comment (EdenWuyifan) [reply voluncheer/voluncheer/settings.py:1]: Yes, those are necessary. There are accessibility issues when uploading image locally, so I debugged it

File: voluncheer/opportunityboard/test_forms.py (hunk 5 above)
Comment (EdenWuyifan) [reply voluncheer/opportunityboard/test_forms.py:43]: image

Void function can only be tested using assertFalse

File: voluncheer/opportunityboard/tests_views.py (hunk 1 above)
Comment (EdenWuyifan) [reply voluncheer/opportunityboard/tests_views.py:21]: 302 error does not mean webpage couldn’t be loaded. It's like too many redirections or so... Hence I omit it

File: voluncheer/opportunityboard/models.py
Comment (EdenWuyifan) [reply voluncheer/opportunityboard/models.py:1]: Sure thing. Feel free to over write that with your branch~

**Review by dbleek — APPROVED** [review]
